src/rpc/blockchain.cpp

```diff
@@ -453,6 +453,12 @@ static RPCHelpMan getblockfrompeer()
         throw JSONRPCError(RPC_MISC_ERROR, "Block header missing");
     }
 
+    // Fetching blocks before the node has syncing past their height can prevent block files from
+    // being pruned, so we avoid it if the node is in prune mode.
+    if (index->nHeight > chainman.ActiveChain().Tip()->nHeight && node::fPruneMode) {
```

Comment:
In 7fa851fb: (non-blocking nit)

Why not use `IsBlockPruned` instead?

Which should be the same as saying that, on pruning mode, can only fetch blocks that were downloaded and discarded.

Reply:
If `IsBlockPruned` was used instead, fetching an older block that isn't pruned (blocks close to the genesis for instance) on a pruned node would result in the error `In prune mode, only blocks that the node has already synced previously can be fetched from a peer` instead of `Block already downloaded`.

Reply:
Doesn't `chainman.ActiveChain()` need to be called with `cs_main` locked? 
```
    CChain& ActiveChain() const EXCLUSIVE_LOCKS_REQUIRED(GetMutex()) { return ActiveChainstate().m_chain; }

```

Reply:
Hmm, indeed it does. I really need to be building clang...

```
rpc/blockchain.cpp:464:35: warning: calling function 'ActiveChain' requires holding mutex 'cs_main' exclusively [-Wthread-safety-analysis]
    if (index->nHeight > chainman.ActiveChain().Tip()->nHeight && node::fPruneMode) {
                                  ^
1 warning generated.
```